libqtile/backend/x11/drawer.py

```diff
@@ -88,20 +112,27 @@ def _free_gc(self):
                 self.qtile.core.conn.conn.core.FreeGC(self._gc)
             self._gc = None
 
-    def _create_xcb_surface(self):
+    def _create_xcb_surface(self, pixmap=None):
+        pixmap = self._pixmap if pixmap is None else pixmap
         surface = cairocffi.XCBSurface(
             self.qtile.core.conn.conn,
-            self._pixmap,
+            pixmap,
             self._visual,
             self.width,
             self.height,
         )
         return surface
 
-    def _free_xcb_surface(self):
-        if self._xcb_surface is not None:
-            self._xcb_surface.finish()
-            self._xcb_surface = None
+    def _free_surfaces(self):
+        for surface in [
+            self._xcb_surface,
+            self.surface,
+            self._pseudo_surface,
+            self._root_surface,
+        ]:
+            if surface is not None:
+                surface.finish()
+                surface = None
```

Comment:
`surface = None` here and `pixmap = None` on line 153 are redundant - the variables get rebound immediately after at the start of the next loop. If you mean to reset the attributes you might need to make it loop over attribute names and use `getattr`/`setattr`